core/include/drivers/stm32_rif.h

```diff
@@ -108,57 +114,83 @@ static inline TEE_Result stm32_rifsc_check_tdcid(bool *tdcid_state)
  * -When CID filtering is enabled on a controller, semaphore
  *  is enabled and is taken, and Cortex A35 CID is
  *  white-listed and semaphore is already taken by Cortex A35
+ *
+ * @cidcfgr:		CIDCFGR configuration register value
+ * @semcr:		SEMCR register value
+ * @nb_cid_supp:	Number of CID supported
+ * @cid_to_check:	CID to check access for
  */
-#ifdef CFG_STM32_RIF
 TEE_Result stm32_rif_check_access(uint32_t cidcfgr,
 				  uint32_t semcr,
 				  unsigned int nb_cid_supp,
 				  unsigned int cid_to_check);
-#else
-static inline TEE_Result
-		stm32_rif_check_access(uint32_t cidcfgr __unused,
-				       uint32_t semcr __unused,
-				       unsigned int nb_cid_supp __unused,
-				       unsigned int cid_to_check __unused)
-{
-	return TEE_SUCCESS;
-}
-#endif
 
 /*
- * Parse RIF config from Device Tree extracted information
+ * stm32_rif_parse_cfg - Parse RIF config from Device Tree extracted information
+ *
  * @rif_conf: Configuration read in the device tree
  * @conf_data: Buffer containing the RIF configuration to apply for an IP
- * @sec_conf: Buffer containing the secure RIF configuration for an IP
- * @priv_conf: Buffer containing the CID RIF configurations for an IP channels
- *             (i.e: GPIO pins, FMC controllers)
  * @nb_cid_supp: Number of supported CID for the IP
  * @nb_channel: Number of channels for the IP
  */
-#ifdef CFG_STM32_RIF
 void stm32_rif_parse_cfg(uint32_t rif_conf,
 			 struct rif_conf_data *conf_data,
 			 unsigned int nb_cid_supp,
 			 unsigned int nb_channel);
-#else
-static inline void
-		stm32_rif_parse_cfg(uint32_t rif_conf __unused,
-				    struct rif_conf_data *conf_data __unused,
-				    unsigned int nb_cid_supp __unused,
-				    unsigned int nb_channel __unused)
-{
-}
-#endif
 
-#ifdef CFG_STM32_RIF
-/* RIF semaphore functions */
-bool stm32_rif_is_semaphore_available(vaddr_t addr);
+/*
+ * stm32_rif_semaphore_is_available() - Checks if the _SEMCR_MUTEX bit is set
+ *
+ * @addr:	Address of the register to read from
```

Comment:
Prefer use a single space char instead of the inner inner tab.
Ditto below

(but that's not a strong opinion)